src/main/java/com/powsybl/openloadflow/network/impl/LfLoads.java

```diff
@@ -14,69 +14,91 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
 public class LfLoads extends AbstractElement {
 
     private final List<Load> loads = new ArrayList<>();
 
-    private final List<Double> participationFactors = new ArrayList<>();
+    private double[] participationFactors;
 
-    private final List<Double> powerFactors = new ArrayList<>();
```

Comment:
Why have you removed that 2 tables `powerFactors` and `p0s`? As you have done, it is similar as what I have done before and with the comment that we don't have to expose IIDM network in the LfNetwork. The method `getLoadTargetQ` is used in a outerloop.

Reply:
Because there's no need to store them, they're already stored in the loads, and remain constant. What you did before was exposing IIDM network by adding `getLoads` to `Lfbus` interface, which was returning `Load` IIDM objects, and you were handling these loads in the distribution step itself. Here IIDM is not exposed, it stays within `LfLoads`.